src/modules/adaptors/Frax/DebtFTokenAdaptorV2.sol

```diff
@@ -0,0 +1,300 @@
+     * @notice For mainnet use 0x853d955aCEf822Db058eb8505911ED77F175b99e.
+     */
+    ERC20 public immutable FRAX;
+
+    /**
+     * @notice This bool determines how this adaptor accounts for interest.
+     *         True: Account for pending interest to be paid when calling `balanceOf` or `withdrawableFrom`.
+     *         False: Do not account for pending interest to be paid when calling `balanceOf` or `withdrawableFrom`.
+     */
+    bool public immutable ACCOUNT_FOR_INTEREST;
+
+    /**
+     * @notice Minimum Health Factor enforced after every borrow.
+     * @notice Overwrites strategist set minimums if they are lower.
+     */
+    uint256 public immutable minimumHealthFactor;
+
+    constructor(bool _accountForInterest, address _frax, uint256 _healthFactor) {
+        _verifyConstructorMinimumHealthFactor(_healthFactor);
+        ACCOUNT_FOR_INTEREST = _accountForInterest;
+        FRAX = ERC20(_frax);
+        minimumHealthFactor = _healthFactor;
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("FraxLend debtToken Adaptor V 1.0"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+
+    /**
+     * @notice User deposits are NOT allowed into this position.
+     */
+    function deposit(uint256, bytes memory, bytes memory) public pure override {
+        revert BaseAdaptor__UserDepositsNotAllowed();
+    }
+
+    /**
+     * @notice User withdraws are NOT allowed from this position.
+     */
+    function withdraw(uint256, address, bytes memory, bytes memory) public pure override {
+        revert BaseAdaptor__UserWithdrawsNotAllowed();
+    }
+
+    /**
+     * @notice This position is a debt position, and user withdraws are not allowed so
+     *         this position must return 0 for withdrawableFrom.
+     */
+    function withdrawableFrom(bytes memory, bytes memory) public pure override returns (uint256) {
+        return 0;
+    }
+
+    /**
+     * @notice Returns the cellar's balance of the respective Fraxlend debtToken calculated from cellar borrow shares
+     * @param adaptorData encoded fraxlendPair (fToken) for this position
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        IFToken fraxlendPair = abi.decode(adaptorData, (IFToken));
+        return _toBorrowAmount(fraxlendPair, fraxlendPair.userBorrowShares(msg.sender), false, ACCOUNT_FOR_INTEREST);
```

Comment:
Use internal function instead of directly calling fraxlendPair. `_userBorrowShares` can have an extra address added so that this function can pass in `msg.sender`, but strategist functions can pass in `address(this)`

Reply:
`_userBorrowShares()` returns the `borrowShares` not the `borrowAmount`, so I used `_toBorrowAmount()` with the `borrowShares` reported from `_userBorrowShares()`. I've implemented the change in `DebtFTokenAdaptorV1` too.

It is generating this error though, now within multiple tests.
<img width="1238" alt="image" src="https://github.com/PeggyJV/cellar-contracts/assets/131093442/3a20ec43-5fe4-441c-8ffe-00fce9cda1ee">

Reply:
Made adjustments as per our discussion, now it works!